app/src/main/java/com/poupa/vinylmusicplayer/preferences/SharedPreferencesExporter.java

```diff
@@ -0,0 +1,97 @@
+package com.poupa.vinylmusicplayer.preferences;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.ParcelFileDescriptor;
+
+import com.google.gson.Gson;
+import com.poupa.vinylmusicplayer.R;
+import com.poupa.vinylmusicplayer.util.OopsHandler;
+import com.poupa.vinylmusicplayer.util.SafeToast;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceManager;
+
+public class SharedPreferencesExporter extends AppCompatActivity {
+    private Context context;
+    private static final String FILENAME = "filename";
+    private ActivityResultLauncher<String> exportFilePicker;
+    private SharedPreferences sharedPreferences;
+
+    public static void start(Context context, String filename) {
+        Intent intent = new Intent(context, SharedPreferencesExporter.class);
+        intent.putExtra(FILENAME, filename);
+        context.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.context = this.getApplicationContext();
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this.context);
+        Bundle b = this.getIntent().getExtras();
+        exportFilePicker = registerForActivityResult(new ActivityResultContracts.CreateDocument("text/plain"), result -> {
+            // Unless the selection has been cancelled, create the export file
+            if(result != null) {
+                try {
+                    writeToExportFile(result);
+                } catch (PackageManager.NameNotFoundException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            // Finishes the last activity to return to the settings activity.
+            this.finish();
+        });
+        exportFilePicker.launch(b.getString(FILENAME));
+    }
+
+    private void writeToExportFile(Uri location) throws PackageManager.NameNotFoundException {
+        Gson gson = new Gson();
+        HashMap<String, Object> prefsMap = new HashMap<>(sharedPreferences.getAll());
+        HashMap<String, Object> copyMap = new HashMap<>(prefsMap);
+        List<String> prefsFilter = Arrays.asList("SONG_IDS_");;
```

Comment:
For easier maintenance, please consider the following suggestions (in increasing degree of complexiity):
- I'd promote this list as a class static constant, i.e. make it more visible/discoverable than being just a local variable here.
- Or name all the non-exportable prefs as `xxx_NON_EXPORTABLE`, hence no need for a manually-maintained list.
- Or Ideally, we should have a decorator such as `@VinylPrefsNotExportable` that we attach to the prefs declaration. And here use that decorator to pick or not the prefs for exporting

Reply:
Can you please exclude as well the following pref keys from being exported as well:
- `OOPS_HANDLER_EXCEPTIONS` : This stores captured crash dumps
- `NEXT_SLEEP_TIMER_ELAPSED_REALTIME`: the ongoing sleep timer
- `LAST_CHANGELOG_VERSION` and `INTRO_SHOWN`: related to intro screen
- `INITIALIZED_BLACKLIST`: related to one-time setup that will always run on a reinstall
- `SAF_SDCARD_URI`: related to one-time setup, involving user permission UI
- `SAVED_POSITION` and `SAVED_POSITION_IN_TRACK`: related to current playing song
- `PREF_MIGRATED_STATIC_PLAYLISTS`: related to one-time setup upon reinstall
- `CUSTOM_ARTIST_IMAGE_PREFS` and `ARTIST_SIGNATURE_PREFS`: related to artist cache

Reply:
Given this extended list, I would like to attempt the decorator/annotation mentioned above.
How about I'm creating a branch to introduce those decoration/annotations first, and you can change this PR to point to that branch instead?

Reply:
The annotation idea is a good one I think. Is your idea to add the annotation and then the exporter should filter them automatically without the need of adding them into a filter? If so, I'd rather postpone adding the pref keys as you've mentioned above.
Let me know the branch if you have created one, then I try to point my branch to it.

Reply:
Hi, I've outlined the annotation idea here: https://github.com/VinylMusicPlayer/VinylMusicPlayer/compare/pref-annotation?expand=1

Feel free to rework the `PreferenceUtil.checkAnnotations` function, or remove it - I've created this new function to illustrate how the annotations can be used.

Would this work for you?

Reply:
Thanks for creating the annotation approach:

One thought on your work:
- What do you think about defaulting [ExportImportable ](https://github.com/VinylMusicPlayer/VinylMusicPlayer/compare/pref-annotation?expand=1#diff-3050ba5dc9c4a15b9e110f24cad42a229e413fb2d9bce2d7681807db02387a14R11) to true as this is probably more often the case than false. Or is this in general a bad approach?

Btw. I've also pointed my PR to pref-annotation.

Reply:
> * What do you think about defaulting ExportImportable to true as this is probably more often the case than false. Or is this in general a bad approach?

Dont hesitate to do that, the branch is for our experiment.

Or you can change the PrefKey class to ExportImportablePref. Initially I've intended this annotation for all pref keys, but for now there is no use of non-exportable annotations